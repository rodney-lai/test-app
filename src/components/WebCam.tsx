/**
 *
 * Copyright (c) 2020 Rodney S.K. Lai
 * https://github.com/rodney-lai
 *
 * Permission to use, copy, modify, and/or distribute this software for
 * any purpose with or without fee is hereby granted, provided that the
 * above copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 */

import * as React from 'react'
import { useState, useEffect } from 'react'
import Typography from '@material-ui/core/Typography'

const WebCam = (props: {show: boolean}) => {
  const [info, setInfo] = useState({dateCreated:null})

  useEffect(() => {
    if (props.show) {
      fetch("https://img.rodneylai.com/webcam.json?ticks=" + (new Date()).getTime())
        .then(result => result.json())
        .then(result => setInfo(result))
    }
  }, [props.show])

  if (props.show) {
    return(
      <>
        <div><Typography variant="subtitle1">{info.dateCreated}</Typography></div>
        <div>
          <img src={"https://img.rodneylai.com/webcam.jpg?ticks=" + (new Date()).getTime()} alt="webcam" title="webcam" width="95%"/>
        </div>
      </>
    )
  } else {
    return null
  }
}

export default WebCam
