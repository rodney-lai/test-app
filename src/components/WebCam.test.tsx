/**
 *
 * Copyright (c) 2020 Rodney S.K. Lai
 * https://github.com/rodney-lai
 *
 * Permission to use, copy, modify, and/or distribute this software for
 * any purpose with or without fee is hereby granted, provided that the
 * above copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 */

import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import { rest } from 'msw'
import { setupServer } from 'msw/node'
import WebCam from './WebCam'

const server = setupServer(
  rest.get('https://img.rodneylai.com/webcam.json', (req, res, ctx) => {
    return res(ctx.json({
      "key":"theKey",
      "dateCreated":"theDateCreated"
    }))
  })
)

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())

const now = 1599956873663

test('webcam component when show = true', async () => {
  jest
    .spyOn(global.Date, 'now')
    .mockImplementation(() =>
      new Date(now).valueOf()
    )
  const { container } = render(<WebCam show={true} now={now} />)
  await waitFor(() => screen.getByText(/theDateCreated/i))
  expect(container).toMatchSnapshot()
})

test('webcam component when show = false', () => {
  const { container } = render(<WebCam show={false} now={now} />)
  expect(container).toMatchSnapshot()
})
