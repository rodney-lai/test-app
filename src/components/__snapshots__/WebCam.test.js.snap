// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webcam component when show = false 1`] = `<div />`;

exports[`webcam component when show = true 1`] = `
<div>
  <div>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1"
    >
      theDateCreated
    </h6>
  </div>
  <div>
    <img
      alt="webcam"
      src="https://img.rodneylai.com/webcam.jpg?ticks=1599956873663"
      title="webcam"
      width="95%"
    />
  </div>
</div>
`;
